applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.cpp

```diff
@@ -300,7 +300,9 @@ void AdjointFiniteDifferencingBaseElement::CalculateSensitivityMatrix(const Vari
 {
     KRATOS_TRY;
 
-    const double delta = this->GetPerturbationSize(rDesignVariable);
+    // TODO Mahmoud: The perturbation value in GetPerturbationSize() with displacement
+    // should fix GetPerturbationSize()function
+    const double delta = this->GetValue(PERTURBATION_SIZE); 
```

Comment:
this is an elemental-value?

Reply:
It is actually a constant value defined in the input json file but for some reason, they defined it as an element value, and they get the value by function GetPerturbationSize() but there is something buggy there as the perturbation value change over the time steps

Reply:
then you should define it in the processinfo

after #2993 there is no way any more that it can change unnoticed